core/src/executor/evaluate/function.rs

```diff
@@ -501,6 +501,25 @@ pub fn prepend<'a>(expr: Evaluated<'_>, value: Evaluated<'_>) -> Result<Evaluate
     }
 }
 
+pub fn take<'a>(expr: Evaluated<'_>, size: Evaluated<'_>) -> Result<Evaluated<'a>> {
```

Comment:
you can consider to get `name` param like other functions.
Then the `name` param can be used in `EvaluateError::FunctionRequiresUSizeValue(name)` like errors.